### PR TITLE
fix: special characters messing with rendering

### PR DESCRIPTION
**Pull request by YannickPferr**
There are some special characters that are not accounted for in the go-runewidth package we are using to determine the width of a string. This leads to the string width being miscalculated when having these characters in there and thus the cursor being set to an incorrect position. This then leads to weird indentations in the output window.

We could solve this by just using string len instead of the runewidth package but then we lose support for Chinese characters, emojis and possibly other characters. Even though this might not be too relevant anyways we decided to keep the package and rather sanitize the input before inserting it into the buffer. This means finding and removing all occurrences of ASCII control sequences.

The challenging part about removing those sequences is that there are sequences that are subsequences of other sequences. To avoid replacing only part of a sequence we sort the sequences by length and then find and replace. This should cover every case since every sequence that is longer than one character starts with the escape char and thus there are no sequences starting with the ending char of another sequence (that basically means, it's not possible to construct a sequence as a linear combination of two or more other sequences). 

TLDR: to avoid weird rendering we sanitize the input from ASCII control sequences before inserting it into the buffer.

### CODE REVIEW

File: input_test.go [hunk 1]
```diff
@@ -25,9 +27,30 @@ func TestPosixParserGetKey(t *testing.T) {
...
-				t.Errorf("Should be %s, but got %s", key, s.expected)
-			}
+			assert.Equal(t, s.expected, key)
 		})
 	}
 }
+
+func RandomASCIIByteSequence() *rapid.Generator[[]byte] {
+	return rapid.Custom(func(t *rapid.T) []byte {
+		return rapid.SampledFrom(ASCIISequences).Draw(t, "random ascii sequence").ASCIICode
```
Comment (gustavodemorais) [inline input_test.go:37]: Nice! Looks good 💯

**Review by gustavodemorais — APPROVED** [review]
